imap/http_caldav_sched.c

```diff
@@ -725,22 +725,31 @@ static int imip_send(const char *cal_ownerid, const char *sched_userid,
     const char *imip_method = icalproperty_method_to_string(
                                   icalcomponent_get_method(sched_data->itip));
 
-    json_t *val = json_pack("{s:s s:s s:s s:s s:o s:s s:o s:o s:b s:s}",
-                            "calendarOwner", cal_ownerid,
-                            "recipient", buf_cstring(&recipient),
-                            "sender", sender,
-                            "method", imip_method,
-                            "id", id,
-                            "ical", ical_str,
-                            "jsevent", jsevent,
-                            "patch", patch,
-                            "is_update", SCHED_IS_UPDATE(sched_data),
-                            "schedulingMechanism",
-                            sched_mechanisms[sched_data->mech]);
-    char *serial = json_dumps(val, JSON_COMPACT);
-    notify(notifier, "IMIP", NULL, sched_userid, NULL, 0, NULL, serial, NULL);
-    free(serial);
-    json_decref(val);
+    json_error_t jerr;
+    json_t *val = json_pack_ex(&jerr, 0,
+                               "{s:s s:s s:s s:s s:o s:s s:o s:o s:b s:s}",
+                               "calendarOwner", cal_ownerid,
+                               "recipient", buf_cstring(&recipient),
+                               "sender", sender,
+                               "method", imip_method,
+                               "id", id,
+                               "ical", ical_str,
+                               "jsevent", jsevent,
+                               "patch", patch,
+                               "is_update", SCHED_IS_UPDATE(sched_data),
+                               "schedulingMechanism",
+                               sched_mechanisms[sched_data->mech]);
+    if (!val) {
+        xsyslog(LOG_ERR, "failed to create iMIP notification message",
```

Comment:
Should we set r to something in this case so the error bubbles up, or no? We don't above if  icalcomponent_as_ical_string fails, so maybe that's okay?

Reply:
Yeah, I'll do that

Reply:
(We should probably add a log line for when icalcomponent_as_ical_string fails too)